web/toolbar.js

```diff
@@ -0,0 +1,282 @@
+ * @property {HTMLButtonElement} previous - Button to go to the previous page.
+ * @property {HTMLButtonElement} next - Button to go to the next page.
+ * @property {HTMLButtonElement} zoomIn - Button to zoom in the current page.
+ * @property {HTMLButtonElement} zoomOut - Button to zoom out the current page.
+ * @property {HTMLButtonElement} viewFind - Button to open find bar.
+ * @property {HTMLButtonElement} openFile - Button to for opening file dialog.
+ * @property {HTMLButtonElement} presentationModeButton - Button to switch to
+ *   the presentation mode.
+ * @property {HTMLButtonElement} download - Button to download the document.
+ * @property {HTMLAElement} viewBookmark - Element to link current url of
+ *   the page view.
+ */
+
+/**
+ * @class
+ */
+var Toolbar = (function ToolbarClosure() {
+  /**
+   * @constructs Toolbar
+   * @param {ToolbarOptions} options
+   * @param {HTMLDivElement} mainContainer
+   * @param {EventBus} eventBus
+   */
+  function Toolbar(options, mainContainer, eventBus) {
+    this.toolbar = options.container;
+    this.mainContainer = mainContainer;
+    this.eventBus = eventBus;
+    this.items = options;
+
+    this._wasLocalized = false;
+    this.reset();
+
+    // Bind the event listeners for click and hand tool actions.
+    this._bindListeners();
+  }
+
+  Toolbar.prototype = {
+    setPageNumber: function (pageNumber, pageLabel) {
+      this.pageNumber = pageNumber;
+      this.pageLabel = pageLabel;
+      this._updateUIState(false);
+    },
+
+    setPagesCount: function (pagesCount, hasPageLabels) {
+      this.pagesCount = pagesCount;
+      this.hasPageLabels = hasPageLabels;
+      this._updateUIState(true);
+    },
+
+    setPageScale: function (pageScaleValue, pageScale) {
+      this.pageScaleValue = pageScaleValue;
+      this.pageScale = pageScale;
+      this._updateUIState(false);
+    },
+
+    reset: function () {
+      this.pageNumber = 0;
+      this.pageLabel = null;
+      this.hasPageLabels = false;
+      this.pagesCount = 0;
+      this.pageScaleValue = DEFAULT_SCALE_VALUE;
+      this.pageScale = DEFAULT_SCALE;
+      this._updateUIState(true);
+    },
+
+    _bindListeners: function Toolbar_bindClickListeners() {
+      var eventBus = this.eventBus;
+
+      this.items.previous.addEventListener('click', function() {
+        eventBus.dispatch('nextpage');
+      });
+
+      this.items.next.addEventListener('click', function() {
+        eventBus.dispatch('previouspage');
+      });
+
+      this.items.zoomIn.addEventListener('click', function() {
+        eventBus.dispatch('zoomin');
+      });
+
+      this.items.zoomOut.addEventListener('click', function() {
+        eventBus.dispatch('zoomout');
+      });
+
+      this.items.pageNumber.addEventListener('click', function() {
+        this.select();
+      });
+
+      this.items.pageNumber.addEventListener('change', function() {
+        eventBus.dispatch('pagenumberchanged', {value: this.value});
```

Comment:
Shouldn't the event also include the `source` property, given that all other usages of the `EventBus` does? I.e.
```javascript
eventBus.dispatch('pagenumberchanged', {
  source: self, // With an added `var self = this;` at the top of the method, since you probably can't use `bind(this)` here. 
  value: this.value,
});`